src/test/java/reposense/parser/ArgsParserTest.java

```diff
@@ -0,0 +1,174 @@
+package reposense.parser;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.Optional;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import reposense.exception.ParseException;
+import reposense.frontend.CliArguments;
+import reposense.util.TestUtil;
+
+public class ArgsParserTest {
+
+    private static final String INPUT_DELIMITER = " ";
+    private static final String PROJECT_DIRECTORY = System.getProperty("user.dir");
+
+    private static final String CONFIG_FILE_ABSOLUTE = Paths.get(PROJECT_DIRECTORY, "sample_small.csv").toString();
+    private static final String OUTPUT_DIRECTORY_ABSOLUTE = Paths.get(PROJECT_DIRECTORY, "build").toString();
+    private static final String CONFIG_FILE_RELATIVE = Paths.get("sample_small.csv").toString();
+    private static final String OUTPUT_DIRECTORY_RELATIVE = Paths.get("build").toString();
+    private static final String DEFAULT_MANDATORY_ARGS = "-config " + CONFIG_FILE_ABSOLUTE + " ";
+
+    @Test
+    public void parse_allCorrectInputs_success() throws ParseException {
+        String input = DEFAULT_MANDATORY_ARGS + String.format("-output %s -since 01/07/2017 -until 30/11/2017",
```

Comment:
Since this portion is also testing for the `config` settings, should add it the `String.format` instead of using `DEFAULT_MANDATORY_ARGS`